server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/subsystem/ProtocolServerMetricsHandler.java

```diff
@@ -84,6 +53,10 @@ protected void executeRuntimeStep(OperationContext context, ModelNode operation)
          case BYTES_WRITTEN:
             result.set(service.getTransport().getTotalBytesWritten());
```

Comment:
you need to get the `Transport` from the `AbstractProtocolServer`. The service, only keeps the first instance instantiated by the endpoint.

Reply:
Done